src/utils/workflow_utils/src/workflow_utils.c

```diff
@@ -2157,35 +2408,56 @@ ADUC_Result _workflow_init_helper(ADUC_WorkflowHandle* handle)
     return result;
 }
 
-ADUC_Result workflow_init_from_file(const char* updateManifestFile, bool validateManifest, ADUC_WorkflowHandle* handle)
+ADUC_Result
+workflow_init_from_file(const char* updateManifestFile, bool validateManifest, ADUC_WorkflowHandle* outWorkflowHandle)
 {
-    ADUC_Result result = { ADUC_GeneralResult_Failure };
-    if (updateManifestFile == NULL || handle == NULL)
+    ADUC_Result result = { .ResultCode = ADUC_GeneralResult_Failure, .ExtendedResultCode = 0 };
+
+    ADUC_WorkflowHandle workflowHandle = NULL;
+
+    JSON_Value* rootJsonValue = NULL;
+
+    if (updateManifestFile == NULL || outWorkflowHandle == NULL)
     {
         result.ExtendedResultCode = ADUC_ERC_UTILITIES_WORKFLOW_UTIL_ERROR_BAD_PARAM;
         goto done;
     }
 
-    result = _workflow_parse(true, updateManifestFile, validateManifest, handle);
+    result = workflow_parse_json(true /* isFile */, updateManifestFile, &rootJsonValue);
```

Comment:
Should set *outWorkflowHandle = NULL before line 2426

Reply:
fixed